### PR TITLE
Dev pga

### PR DESCRIPTION
**Pull request by pgaliano**

### CODE REVIEW

File: core/IEnviroment.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        string GetVariable(string name);
+    }
+
+    public class Enviroment : IEnviroment
+    {
+        public string GetVariable(string name)
+        {
+            Exceptions.ThrowIfNullOrEmpty(name, nameof(name));
+
+            return Environment.GetEnvironmentVariable(name);
```
Comment (kzu) [inline core/IEnviroment.cs:16]: a more suscint way of doing this that doesn't require the "old-school" `Exceptions.ThrowIfNullOrEmpty` is two-fold: enable `<Nullable>Enable</Nullable>` in the project (therefore not needing to check for null), and then maybe an `Ensure.NotEmpty(name, nameof(name))` which returns de value or throws. 
With that in place, this implementation would just be:

```csharp
public string GetVariable(string name) => Environment.GetEnvironmentVariable(Ensure.NotEmpty(name, nameof(name)));
```

File: models/IncomingMessage.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using Newtonsoft.Json.Linq;
+
+namespace NosAyudamos
+{
+    public class IncomingMessage
```
Comment (kzu) [inline models/IncomingMessage.cs:5]: Is this really an *incoming*? I feel like both incoming and outgoing messages are going to have the same three properties at least.

File: services/IDonor.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        public State State { get; set; }
+
+        public Donor()
+        {
+            _machine = new StateMachine<State, Trigger>(() => State, s => State = s);
+            InitializeStateMachine();
+        }
+
+        public void Register()
```
Comment (kzu) [inline services/IDonor.cs:23]: I like one-liner members to just use lambdas, like `public void Register() => _machine.Fire(Trigger.Register);`

File: services/IDonor.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public interface IDonor
+    {
+        State State { get; set; }
+        void Register();
+    }
+
+    public class Donor : IDonor
+    {
+        private readonly StateMachine<State, Trigger> _machine;
```
Comment (kzu) [inline services/IDonor.cs:13]: No underscore prefix plz. We need a .editorconfig that will ensure some conventions for all of us :)

File: api.csproj [hunk 5]
```diff
@@ -4,8 +4,12 @@
     <AzureFunctionsVersion>v3</AzureFunctionsVersion>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime" Version="3.0.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.0.0" />
     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.*" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.*" />
+    <PackageReference Include="newtonsoft.json" Version="12.0.3" />
```
Comment (kzu) [inline api.csproj:11]: Let's not use Json.NET anymore. .NETCore has its own new thing in `System.Text.Json` which is tunned for performance and was actually mostly done (I think) by the same James Newton-King ;)

File: services/IDonor.cs (hunk 3 above)
Comment (pgaliano) [reply services/IDonor.cs:23]: will not be on line member, implementation is not complete

**Comment by kzu** [comment]
Don't get heated up :P

**Pull request closed without merging**